### PR TITLE
Stylize `Requires-Python` consistently in CLI output

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: crates/uv/src/commands/project/lock.rs [hunk 1]
```diff
@@ -223,13 +223,13 @@ pub(super) async fn do_lock(
         if requires_python.is_unbounded() {
             let default =
                 RequiresPython::greater_than_equal_version(&interpreter.python_minor_version());
-            warn_user!("The workspace `requires-python` field does not contain a lower bound: `{requires_python}`. Set a lower bound to indicate the minimum compatible Python version (e.g., `{default}`).");
+            warn_user!("The workspace `Requires-Python` field does not contain a lower bound: `{requires_python}`. Set a lower bound to indicate the minimum compatible Python version (e.g., `{default}`).");
```
Comment (konstin) [inline crates/uv/src/commands/project/lock.rs:226]: The problem is that toml is case sensitive, so `Requires-Python = ">=3.8"` does not.

Comment (charliermarsh) [reply crates/uv/src/commands/project/lock.rs:226]: @konstin is probably right on that one.

Comment (zanieb) [reply crates/uv/src/commands/project/lock.rs:226]: Tragic.

Comment (charliermarsh) [reply crates/uv/src/commands/project/lock.rs:226]: I know, aesthetically I dislike it.

Comment (zanieb) [reply crates/uv/src/commands/project/lock.rs:226]: This is a far more invasive change if we want to be consistent throughout our internal comments as well.

**Comment by zanieb** [comment]
I made the minimal changes to the user output but didn't touch our internal commentary.

**Review by charliermarsh — APPROVED** [review]

File: crates/uv/src/commands/python/pin.rs [hunk 2]
```diff
@@ -271,7 +271,7 @@ fn assert_pin_compatible_with_project(pin: &Pin, virtual_project: &VirtualProjec
     };
 
     Err(anyhow::anyhow!(
-        "The {given} Python version `{}`{resolved} is incompatible with the {} `Requires-Python` requirement of `{}`.",
+        "The {given} Python version `{}`{resolved} is incompatible with the {} `requires-python` field of `{}`.",
```
Comment (charliermarsh) [inline crates/uv/src/commands/python/pin.rs:274]: "value", for consistency above?

Comment (charliermarsh) [reply crates/uv/src/commands/python/pin.rs:274]: Defer to you though

**Pull request merged**